cargo-shuttle/src/main.rs

```diff
@@ -51,8 +61,8 @@ impl Shuttle {
             Command::Deploy(deploy_args) => self.deploy(deploy_args).await,
             Command::Status => self.status().await,
             Command::Delete => self.delete().await,
-            Command::Auth(auth_args) => self.auth(auth_args).await,
-            Command::Login(login_args) => self.login(login_args).await,
+            Command::Auth(auth_args) => self.auth(auth_args).await.map(|_| DeploymentStateMeta::Deployed),
+            Command::Login(login_args) => self.login(login_args).await.map(|_| DeploymentStateMeta::Deployed),
```

Comment:
Maybe we should go for something like `CommandStatus` to prevent this unexpected tagging on auth and login?

Reply:
Also, why can't we just use `Result::Err`?